apis/v1alpha2/httproute_types.go

```diff
@@ -315,72 +123,38 @@ type HTTPPathMatch struct {
 // Reason of `UnsupportedValue`.
 //
 // +kubebuilder:validation:Enum=Exact;RegularExpression
-type HeaderMatchType string
+// +k8s:deepcopy-gen=false
+type HeaderMatchType = v1beta1.HeaderMatchType
 
 // HeaderMatchType constants.
 const (
-	HeaderMatchExact             HeaderMatchType = "Exact"
-	HeaderMatchRegularExpression HeaderMatchType = "RegularExpression"
+	HeaderMatchExact             = v1beta1.HeaderMatchExact
```

Comment:
we are not consistent in the way we do conts
here: https://github.com/kubernetes-sigs/gateway-api/pull/1538/files#diff-8f01125c16b3083943cd784b81e7c2b954e37cbfe7b3187a3fa1f5569d16513fR90 we opted for compelete redefinition. but this file opted for redefinition pointing to `v1beta` values. Can we use a single approach across all files?

Reply:
Good catch, I'm removing consts from v1a2 entirely in #1576